### PR TITLE
clarify usage of directions options

### PR DESCRIPTION
**Pull request by ecotner**
clarify use of directions options in top-level of JSON payload, and deprecation of `directions_options` structure

# Issue

Seeks to clarify confusion brought up in #1848 

## Tasklist

 - ~Add tests~
 - ~Add #fixes with the issue number that this PR addresses~
 - [x] Update the docs with any new request parameters or changes to behavior described
 - [x] Update the [changelog](CHANGELOG.md)

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

**Comment by ecotner** [comment]
@kevinkreiser does this look good?

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@
    * FIXED: Allow u-turns at no-access barriers when forced by heading [#2875](https://github.com/valhalla/valhalla/pull/2875)
 
 * **Enhancement**
+   * CHANGED: Clarify usage of directions options [#2886](https://github.com/valhalla/valhalla/pull/2886)
```
Comment (kevinkreiser) [inline CHANGELOG.md:24]: can you move this to the bottom of this enhancement section? i know its wierd but we've been following this pattern

**Review by kevinkreiser — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (ecotner) [reply CHANGELOG.md]: looks like you deleted it and merged; I guess a small edit to the docs is not worth the changelog? I will make sure to do so next time though!

Comment (kevinkreiser) [reply CHANGELOG.md]: yeah at the end of the day it didnt seem necessary, thanks!

**Pull request merged**